esmvaltool/cmorizers/data/formatters/datasets/esacci_sst.py

```diff
@@ -59,7 +63,7 @@ def extract_variable(var_info, raw_info, attrs, year):
             return cube
 
```

Comment:
If we are getting here, it means that the requested variable was not found. Probably we should raise a corresponding error. Better yet, in line 54 we should use `iris.load_cube` with an [`iris.NameConstraint`](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris.html#iris.NameConstraint) instead of the cumbersome loop here. This would also make sure that there isn't a second cube hiding later in the loop.

Reply:
Thanks for your suggestion! Are the changes in line with what you had in mind? Note that the raw data is made of a list of 8 cubes per file. I reran the cmorizer with the changes and output look identical to the ones of the previous cmorizer version.